src/index.js

```diff
@@ -13,34 +26,93 @@ module.exports = new BaseKonnector(start)
 // The start function is run by the BaseKonnector instance only when it got all the account
 // information (fields). When you run this connector yourself in "standalone" mode or "dev" mode,
 // the account information come from ./konnector-dev-config.json file
-function start(fields) {
+async function start(fields) {
+  await authenticate(fields.login, fields.password)
   // The BaseKonnector instance expects a Promise as return of the function
-  return request(`${baseUrl}/index.html`).then($ => {
-    // cheerio (https://cheerio.js.org/) uses the same api as jQuery (http://jquery.com/)
-    // here I do an Array.from to convert the cheerio fake array to a real js array.
-    const entries = Array.from($('article')).map(article =>
-      parseArticle($, article)
-    )
-    return addData(entries, 'io.cozy.books').then(() =>
-      saveFiles(entries, fields)
-    )
+  const $ = await request(`${baseUrl}/index.html`)
+  // cheerio (https://cheerio.js.org/) uses the same api as jQuery (http://jquery.com/)
+  const documents = await parseDocuments($)
+
+  // here we use the saveBills function even if what we fetch are not bills, but this is the most
+  // common case in connectors
+  await saveBills(documents, fields.folderPath, {
```

Comment:
Saving books with `saveBills()` is a bit confusing... Although I understand that books.toscrape.com is convenient... Maybe we should put up our own scrapable two-pages site demo with dummy data at some point? It should not be too hard and we could put it on whatever free node-hosting available out there...

(just thinking out loud)